### ISSUE TITLE
L1 vs scale-and-shift invariant (SSI) loss

### ISSUE THREAD

**Issue by goutamyg** [Eyeline-Research/FlashDepth#10]
Hi,

The paper mentions that it uses _l1_ loss (instead of scale-and-shift invariant loss) during training. Doesn't this force the model to predict (inverse) metric depth and make it sensitive to any global scale and offset errors in the training data? The baseline model depthanythingv2 is trained using SSI loss as well. Curious to know the reason why _l1_ loss leads to better performance/generalization than SSI loss.

The training code has the option to choose SSI loss by setting _loss_type_ as "scaleshift" in the cfg file. However, the forward pass implemented in [train_sequence.py](https://github.com/Eyeline-Research/FlashDepth/blob/main/flashdepth/model.py#L217) seem to reshape the GT and predicted depths as B x T x H x W -> (B x T) x H x W in line numbers [237](https://github.com/Eyeline-Research/FlashDepth/blob/main/flashdepth/model.py#L237) and [234](https://github.com/Eyeline-Research/FlashDepth/blob/main/flashdepth/model.py#L234), respectively, before computing the SSI loss [here](https://github.com/Eyeline-Research/FlashDepth/blob/main/flashdepth/model.py#L244). In this case, the loss function computes scale and shift values on a per-frame basis (i.e., for each of the B x T frames), which looks like a bug to me. Shouldn't the scale and shift be computed on a per-video/global basis during trianing, similar to how it is done while computing the evaluation metrics? 

Goutam

**Comment by gene-chou** [comment]
There wasn't a specific reason for using an L1 loss. All the training datasets are synthetic and metric, so we decided using an L1 loss should be fine. Assuming no scale and offset errors across the datasets, it should effectively be the same as SSI.

As for the second question, we never used SSI for training so I didn't write that code properly. If you were to train with SSI then yes, the scale and shift should be computed per-video.
